### PR TITLE
Add vPortGenerateSimulatedInterruptFromWindowsThread in MSVC port

### PR DESCRIPTION
**Pull request by chinglee-iot**
Add vPortGenerateSimulatedInterruptFromWindowsThread in MSVC port

Description
-----------
In the MSVC port of FreeRTOS, native Windows threads execute concurrently with FreeRTOS tasks and simulated interrupt handlers. Calling a FreeRTOS API from within a native Windows thread can lead to race conditions due to the concurrent execution. One way to enable synchronization between native Windows threads and FreeRTOS tasks is to make use of simulated interrupts. When a native Windows thread needs to invoke a FreeRTOS API, it generates a simulated interrupt and calls the FromISR API within the interrupt service routine (ISR).

The existing `vPortGenerateSimulatedInterrupt` function is designed to be called from within a FreeRTOS task, and it references the current thread structure. To facilitate calling FreeRTOS APIs from native Windows threads, this pull request introduces a new API function, `vPortGenerateSimulatedInterruptFromWindowsThread`. This function allows native Windows threads to generate simulated interrupts and subsequently call FreeRTOS APIs through the FromISR API, ensuring proper synchronization with FreeRTOS tasks.

Existing [tick timer](https://github.com/FreeRTOS/FreeRTOS-Kernel/blob/8e07366994f81354a2d4556ca1da9f73dab781e6/portable/MSVC-MingW/port.c#L140) is an example of generating simulated interrupt from native windows thread.
Another example is the cellular comm interface in this [commit](https://github.com/chinglee-iot/FreeRTOS/commit/e94069b28c528f8f5548d1f5c595ad2490090ffe). Comm interface can generate simulated interrupt to notify FreeRTOS task when UART RX data is received.

In this PR
* Add vPortGenerateSimulatedInterruptFromWindowsThread to enable native windows thread to synchronize with FreeRTOS task through simulated interrupt.

Test Steps
-----------
N/A

Checklist:
----------

- [X] I have tested my changes. No regression in existing tests.
- [ ] ~~I have modified and/or added unit-tests to cover the code changes in this Pull Request.~~

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: portable/MSVC-MingW/port.c [hunk 1]
```diff
@@ -177,28 +177,7 @@ static DWORD WINAPI prvSimulatedPeripheralTimer( LPVOID lpParameter )
...
-             * the handler thread can execute immediately pvInterruptEventMutex is
-             * released. */
-            configASSERT( ulCriticalNesting == 0UL );
-            SetEvent( pvInterruptEvent );
-
-            /* Give back the mutex so the simulated interrupt handler unblocks
-             * and can access the interrupt handler variables. */
-            ReleaseMutex( pvInterruptEventMutex );
-        }
+        vPortGenerateSimulatedInterruptFromNative( portINTERRUPT_TICK );
```
Comment (aggarg) [inline portable/MSVC-MingW/port.c:180]:
```suggestion
        vPortGenerateSimulatedInterruptFromWindowsThread( portINTERRUPT_TICK );
```

File: portable/MSVC-MingW/portmacro.h [hunk 2]
```diff
@@ -184,8 +184,9 @@ void vPortExitCritical( void );
 #define portTASK_FUNCTION_PROTO( vFunction, pvParameters )    void vFunction( void * pvParameters )
 #define portTASK_FUNCTION( vFunction, pvParameters )          void vFunction( void * pvParameters )
 
-#define portINTERRUPT_YIELD    ( 0UL )
-#define portINTERRUPT_TICK     ( 1UL )
+#define portINTERRUPT_YIELD                        ( 0UL )
+#define portINTERRUPT_TICK                         ( 1UL )
+#define portINTERRUPT_APPLICATION_DEFINED_START    ( 2UL )
```
Comment (aggarg) [inline portable/MSVC-MingW/portmacro.h:189]: What is the use of this config?

File: portable/MSVC-MingW/port.c (hunk 1 above)
Comment (chinglee-iot) [reply portable/MSVC-MingW/port.c]: Update in new commit.

File: portable/MSVC-MingW/portmacro.h (hunk 2 above)
Comment (chinglee-iot) [reply portable/MSVC-MingW/portmacro.h:189]: This allows application writer to define interrupt.
For example, in cellular comm interface application
```c
#define portINTERRUPT_UART portINTERRUPT_APPLICATION_DEFINED_START    
```

**Review by aggarg — APPROVED** [review]

**Review by ActoryOu — APPROVED** [review]

**Pull request merged**